libs/barista-components/event-chart/src/event-chart.ts

```diff
@@ -460,15 +589,106 @@ export class DtEventChart<T> implements AfterContentInit, OnInit, OnDestroy {
     return path.join(' ');
   }
 
+  _calculateFieldOutline(
+    renderEvent: RenderField<T>,
+    offset: number = 0,
+  ): string {
+    const eventBubbleRadius = FIELD_BUBBLE_SIZE + offset;
+    const eventBubbleRadiusArcV = FIELD_BUBBLE_SIZE / 2;
+    const eventBubbleRadiusArcH = eventBubbleRadiusArcV + offset / 2;
+    const path: string[] = [];
+    const duration: number =
+      renderEvent.x2 - renderEvent.x1 - FIELD_BUBBLE_SIZE - 2;
+    path.push(`M ${renderEvent.x1},${renderEvent.y - FIELD_BUBBLE_SIZE}`);
+
+    path.push(`h${duration}`);
+
+    path.push(
+      `a${eventBubbleRadiusArcV},${eventBubbleRadiusArcH} 0 0 1 ${eventBubbleRadiusArcV},${eventBubbleRadiusArcH}`,
+    );
+
+    path.push(`v${eventBubbleRadius}`);
+
+    path.push(
+      `a${eventBubbleRadiusArcV},${eventBubbleRadiusArcH} 0 0 1 -${eventBubbleRadiusArcV},${eventBubbleRadiusArcH}`,
+    );
+
+    path.push(`h-${duration}`);
+
+    path.push(
+      `a${eventBubbleRadiusArcV},${eventBubbleRadiusArcH} 0 0 1 -${eventBubbleRadiusArcV},-${eventBubbleRadiusArcH}`,
+    );
+
+    path.push(`v-${eventBubbleRadius}`);
+
+    path.push(
+      `a${eventBubbleRadiusArcV},${eventBubbleRadiusArcH} 0 0 1 ${eventBubbleRadiusArcV},-${eventBubbleRadiusArcH}`,
+    );
+
+    path.push('z');
+
+    return path.join(' ');
+  }
+
+  _calculateRects(renderEvent: RenderField<T>): string {
+    const magic_number = FIELD_BUBBLE_SIZE / 2;
+    const magic_number_2 = magic_number / 2;
```

Comment:
It will be much easier to read the already fairly crazy genius of svg pathing if the variables have at least some semantic meaning. 

```suggestion
    const field_bubble_half = FIELD_BUBBLE_SIZE / 2;
    const field_bubble_quarter = magic_number / 2;
```